layers/pipeline_state.h

```diff
@@ -681,16 +693,23 @@ struct LAST_BOUND_STATE {
 };
 
 static inline bool IsBoundSetCompat(uint32_t set, const LAST_BOUND_STATE &last_bound,
-                                    const PIPELINE_LAYOUT_STATE *pipeline_layout) {
-    bool result = (set < last_bound.per_set.size()) && (set < pipeline_layout->set_compat_ids.size()) &&
-                  (*(last_bound.per_set[set].compat_id_for_set) == *(pipeline_layout->set_compat_ids[set]));
+                                    const PIPELINE_LAYOUT_STATE &pipeline_layout) {
+    // Intentionally have a result variable to simplify debugging
+    bool result = false;
+    if ((set < last_bound.per_set.size()) && (set < pipeline_layout.set_compat_ids.size())) {
+        result = (*(last_bound.per_set[set].compat_id_for_set) == *(pipeline_layout.set_compat_ids[set]));
+    }
     return result;
 }
 
 static inline bool IsPipelineLayoutSetCompat(uint32_t set, const PIPELINE_LAYOUT_STATE *a, const PIPELINE_LAYOUT_STATE *b) {
     // Intentionally have a result variable to simplify debugging
-    bool result = a && b && (set < a->set_compat_ids.size()) && (set < b->set_compat_ids.size()) &&
-                  (a->set_compat_ids[set] == b->set_compat_ids[set]);
+    bool result = false;
+    if (a && b) {
```

Comment:
Avoid nested if statement.

```suggestion
    if (a == nullptr || b == nullptr) { return false; }
```

Reply:
In general I think we avoid the `== nullptr` as we use a lot of smart pointers, and this way we can take advantage of `operator bool` if the type changes.

Reply:
I was under the impression that we were trying to *not* use `== nullptr` for checks like this.

Reply:
I updated the suggestion to reflect that.